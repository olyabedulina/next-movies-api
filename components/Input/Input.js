import React from 'react'
import PropTypes from 'prop-types'

import CM from './styles.module.css'

const Input = ({
    id = '',
    name = '',
    className = '',
    kind = '',
    type = 'text',
    value = '',
    placeholder = '',
    children,
    onChange = Function.prototype,
    onKeyDown = Function.prototype
}) => {

    return <input
        id={id}
        name={name}
        className={`${className} ${CM.input} ${CM["kind-" + kind] || ''}`}
        type={type}
        value={value}
        placeholder={placeholder}
        onChange={onChange}
        onKeyDown={onKeyDown}>
            {children}
        </input>
}

Input.propTypes = {
    id: PropTypes.string,
    name: PropTypes.string,
    className: PropTypes.string,
    kind: PropTypes.string,
    type: PropTypes.string,
    value: PropTypes.node.isRequired,
    placeholder: PropTypes.string,
    children: PropTypes.node,
    onClick: PropTypes.func,
    onKeyDown : PropTypes.func
};

export default Input
