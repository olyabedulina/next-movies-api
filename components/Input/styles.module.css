.input {
  box-sizing: border-box;
  display: inline-block;
  border: 0;
  border-radius: 6px;
  outline: none;
  width: 100%;
  padding: 16px 20px;
  margin: 0;
  font-size: 20px;
  line-height: 20px;
  background: #2e2e2e;
  color: #fff;
}

/*input[type="date"]::-webkit-calendar-picker-indicator {*/
/*  color: rgba(0, 0, 0, 0);*/
/*  opacity: 1;*/
/*  display: block;*/
/*  background: url(./images/calendar.png) no-repeat 0 ;*/
/*  width: 27px;*/
/*  height: 24px;*/
/*  border-width: thin;*/
/*  outline: none;*/
/*  cursor: pointer;*/
/*}*/
