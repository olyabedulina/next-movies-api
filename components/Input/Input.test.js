import React from 'react';
import renderer from 'react-test-renderer';

import Input from './Input';

describe('Components :: Input snapshots:', () => {
    it('Test :: Input type = "text" component', () => {
        const actual = renderer
            .create(<Input
                id='title'
                name='title'
                value={0}
                className="fieldInput"
                placeholder='Title here'
                type='text'
            />)
            .toJSON();
        expect(actual).toMatchSnapshot();
    });

    it('Test :: Input type = "date" component', () => {
        const actual = renderer
            .create(<Input
                id='selectDate'
                name='selectDate'
                className='fieldInput'
                placeholder='Select Date'
                type='date'
                value="20/03/1986"
            />)
            .toJSON();
        expect(actual).toMatchSnapshot();
    });

    it('Test :: Input type = "text" with children component', () => {
        const actual = renderer
            .create(<Input
                id='title'
                name='title'
                value={0}
                className="fieldInput"
                placeholder='Title here'
                type='text'>Text</Input>)
            .toJSON();
        expect(actual).toMatchSnapshot();
    });

    it('Test :: Input type = "text" with children component', () => {
        const actual = renderer
            .create(<Input
                id='checkbox'
                name='checkbox'
                value={0}
                className="fieldCheckbox"
                placeholder='Select genre'
                type='checkbox'>Checkbox</Input>)
            .toJSON();
        expect(actual).toMatchSnapshot();
    });

    it('Test :: Input type = "text" with children component', () => {
        const actual = renderer
            .create(<Input
                id='text'
                name='textInput'
                value='Lorem Ipsum'
                onChange={() => { console.log('test') }}
                className="fieldText"
                placeholder='Select title'
                type='text'>OnChange input</Input>)
            .toJSON();
        expect(actual).toMatchSnapshot();
    });
});
