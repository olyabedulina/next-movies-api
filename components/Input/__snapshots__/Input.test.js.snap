// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components :: Input snapshots: Test :: Input type = "date" component 1`] = `
<input
  className="fieldInput input "
  id="selectDate"
  name="selectDate"
  onChange={[Function]}
  onKeyDown={[Function]}
  placeholder="Select Date"
  type="date"
  value="20/03/1986"
/>
`;

exports[`Components :: Input snapshots: Test :: Input type = "text" component 1`] = `
<input
  className="fieldInput input "
  id="title"
  name="title"
  onChange={[Function]}
  onKeyDown={[Function]}
  placeholder="Title here"
  type="text"
  value={0}
/>
`;

exports[`Components :: Input snapshots: Test :: Input type = "text" with children component 1`] = `
<input
  className="fieldInput input "
  id="title"
  name="title"
  onChange={[Function]}
  onKeyDown={[Function]}
  placeholder="Title here"
  type="text"
  value={0}
>
  Text
</input>
`;

exports[`Components :: Input snapshots: Test :: Input type = "text" with children component 2`] = `
<input
  className="fieldCheckbox input "
  id="checkbox"
  name="checkbox"
  onChange={[Function]}
  onKeyDown={[Function]}
  placeholder="Select genre"
  type="checkbox"
  value={0}
>
  Checkbox
</input>
`;

exports[`Components :: Input snapshots: Test :: Input type = "text" with children component 3`] = `
<input
  className="fieldText input "
  id="text"
  name="textInput"
  onChange={[Function]}
  onKeyDown={[Function]}
  placeholder="Select title"
  type="text"
  value="Lorem Ipsum"
>
  OnChange input
</input>
`;
